src/playlist.js

```diff
@@ -261,12 +261,13 @@ export const duration = function(playlist, endSequence, expired) {
   * playlist in which case, the targetDuration of the playlist is used
   * to approximate the durations of the segments
   *
-  * @param {Object} playlist a media playlist object
-  * @param {number} startIndex
-  * @param {number} endIndex
+  * @param {Object} durationList list to iterate over for durations.
+  * @param {number} defaultDuration duration to use for elements before or after the durationList
+  * @param {number} startIndex partsAndSegments index to start
+  * @param {number} endIndex partsAndSegments index to end.
   * @return {number} the number of seconds between startIndex and endIndex
   */
-export const sumDurations = function(playlist, startIndex, endIndex) {
+export const sumDurations = function({defaultDuration, durationList, startIndex, endIndex}) {
```

Comment:
changes to allow passing in a list and a default duration. This allows us to use `partsAndSegments`, just `parts`, or just `segments`